source/material_model/steinberger.cc

```diff
@@ -453,7 +471,18 @@ namespace aspect
           prm.declare_entry ("Maximum thermal conductivity", "1000",
                              Patterns::Double (0.),
                              "The maximum thermal conductivity that is allowed in the "
-                             "model. Larger values will be cut off.");
+                             "model. Larger values will be cut off."),
+                             prm.declare_entry ("Viscosity averaging scheme", "harmonic",
+                                                Patterns::Selection("arithmetic|harmonic|geometric|maximum composition"),
+                                                "When more than one compositional field is present at a point "
+                                                "with different viscosities, we need to come up with an average "
+                                                "viscosity at that point.  Select a weighted harmonic, arithmetic, "
+                                                "geometric, or maximum composition.");
+          prm.declare_entry ("Prefactors", "1.e21",
+                             Patterns::Anything(),
+                             "List of viscosities for background mantle and compositional fields,"
```

Comment:
Just to be sure, are the "Prefactors" in fact viscosity values, or are they dimensionless composition-dependent factors that the viscosity is multiplied by?

Reply:
Follow up - depending on the answer, I would adjust the parameter name and description to be a bit more specific.

Reply:
Prefactors are the dimensionless composition-dependent factors, for example, if one composition needs to have 10 times higher viscosity than ambient, then prefactor for this composition would be 10.